src/com/model/CashMachine.java

```diff
@@ -0,0 +1,175 @@
+package com.model;
+
+import java.util.Map;
+
+public class CashMachine {
+    // Class with user QNA;
+    UserAnswersScanner userAnswersScanner = new UserAnswersScanner();
+    // Class with banknotes amount and methods to manipulate with them;
+    BanknotesCounter banknotesCounter = new BanknotesCounter();
+    // Banknotes collection;
+    Map<Integer, Integer> banknotesMap = banknotesCounter.banknotesMap;
+    private final int[] ALLOWED_BANKNOTES = banknotesCounter.ALLOWED_BANKNOTES;
+    BanknotesHistory banknotesHistory = new BanknotesHistory(ALLOWED_BANKNOTES);
+
+    // Info about availability of banknotes at CashMachine;
+    public void cache() {
+        for (Map.Entry<Integer, Integer> item : banknotesMap.entrySet()) {
+
+            int banknoteAmount = item.getValue();
+            int banknoteNominal = item.getKey();
+
+            System.out.println("Nominal: " + banknoteNominal + ", Amount " + banknoteAmount);
+        }
+    }
+
+    public void stat() {
+        Map<Integer, BanknotesHistory.BanknoteHistoryData> historyMap = banknotesHistory.getHistoryMap();
```

Comment:
That is working fine but the notation is a bit ugly) 
1) The nested classes are some feature in Java, but frankly speaking, nested classes should be used reasonably. Many of the time, you will have a reviewer telling you "why this class is nested? move it to the upper level because it  reads better when at the upper level!"
2) When it is still justified to use a nested class, give it a name to avoid wordiness. `BanknotesHistory.Data` is way better than `BanknotesHistory.BanknoteHistoryData` because you just repeat repeat yourself)